### ISSUE TITLE
Dropdown options missing IDs

### ISSUE THREAD

**Issue by afercia** [select2/select2#4350]
Note: this issue is part of several steps that should be taken to make the Select2 list options accessible, see #3735 and #4349.

Each item in the results list should always have an unique ID to be used as value for `aria-activedescendant` on the search field (or on the selection element when the search is hidden).

Without `aria-activedescendant` (and `aria-owns` and other aria attributes), assistive technologies can't understand which item in the list is currently "highlighted", thus they can't announce its content.

As far as I see, Select2 is based on various "data adapters" to normalize the data to pass to the widget. See BaseAdapter, SelectAdapter, ArrayAdapter, AjaxAdapter, etc. On the other hand, each item object in the results list is "normalized" by a `_normalizeItem()` function which in turn calls `generateResultId()`

If I'm not wrong, this is the relevant call in `src/js/select2/data/select.js`

```
if (item._resultId == null && item.id && this.container != null) {
  item._resultId = this.generateResultId(this.container, item);
}
```

It seems this fails for some adapters (e.g. array, ajax, tags) because at this point `this.container` is null. See in the screenshot below "red", "blue", and "green" don't have an ID:

![07 missing ids](https://cloud.githubusercontent.com/assets/1682452/15117200/c89acd8e-1606-11e6-9524-d337b9ceb2f8.png)

I'd appreciate any feedback here. Also, I'm not sure I can find a fix without help from someone with more advanced coding abilities than me :) Wondering what to do here, maybe the simpler thing would be using a different way to generate a unique ID?

**Comment by kevin-brown** [comment]
This appears to only happen for items passed in as `data`, making it an issue in the `ArrayAdapter`.
